src/main/java/ch/njol/skript/util/VisualEffect.java

```diff
@@ -159,17 +160,17 @@ public boolean isColorable() {
 		ITEM_CRACK(Particle.ITEM_CRACK) {
 			@Override
 			public Object getData(final @Nullable Object raw, final Location l) {
-				if (raw == null)
-					return Material.IRON_SWORD;
-				else if (raw instanceof ItemType) {
+				ItemStack itemStack = Aliases.javaItemType("iron sword").getRandom();

```

Comment:
I think this could be a ``private final static`` constant for the class

Reply:
If it were, id have to clone it each time its used. In that case its the same as creating a new item stack.
Unless I were to make a final private static ItemType then use getRandom within this method.

Thoughts?

Reply:
https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/expressions/ExprSkull.java I'm not sure, but other components of Skript do it like that

Reply:
Cloning might be marginally cheaper than constructing the ItemType again, not that it really matters on this case though.

Reply:
Im going to leave it as is. As snow said it won't really matter here.